### PR TITLE
Use Buffer or opts.atob if available for atob

### PR DESCRIPTION
**Pull request by goldibex**
Currently stacktrace-gps only uses window.atob for converting base64
encoded inline sourcemaps back to their original representation. This PR
provides a more robust implementation that first looks for an atob
function supplied in the options. By default it first tries window.atob,
then tries Buffer (in nodejs).

### CODE REVIEW

**Comment by goldibex** [comment]
Oy, sorry, I branched this one from `master` (not my own PR) and git made a complete hash of the merge. Will fix that.

**Comment by goldibex** [comment]
Alright, that should do it.

File: spec/stacktrace-gps-spec.js [hunk 1]
```diff
@@ -439,7 +439,161 @@ describe('StackTraceGPS', function () {
...
+            runs(function() {
+                expect(atobspy).toHaveBeenCalled();
+            });
+
+        });
+
+        it('errors if neither window.atob nor Buffer are available', function() {
+
+            var oldatob;
+            var bufferWasCreated = false;
```
Comment (eriwen) [inline spec/stacktrace-gps-spec.js]: This variable isn't used.

File: spec/stacktrace-gps-spec.js [hunk 2]
```diff
@@ -439,7 +439,161 @@ describe('StackTraceGPS', function () {
...
+                server.respond();
+            });
+            waits(100);
+            runs(function() {
+                server.respond();
+            });
+            waits(100);
+            runs(function() {
+                window.atob = oldatob;
+                expect(callback).toHaveBeenCalled();
```
Comment (eriwen) [inline spec/stacktrace-gps-spec.js]: I'd expect assertions here that reflect the test `'errors if neither window.atob nor Buffer are available'`. 

Side note, sorry these tests are a pain to write -- we've been slowly converting to Jasmine 2.x and once we can do that here these will be much simpler. 

Also, yay tests!

File: dist/stacktrace-gps.js [hunk 3]
```diff
@@ -38,6 +38,25 @@
...
 
+    /**
+     * Convert a Base64-encoded string into its original representation.
+     * Used for inline sourcemaps.
+     *
+     * @param b64str [String]
+     * @return The original representation of the base64-encoded string.
+     */
+    function _atob(b64str) {
+
```
Comment (eriwen) [inline dist/stacktrace-gps.js]: Stupid nitpick: could you consider removing the blank lines at the beginning and end of functions. This is consistent with the rest of the codebase.

**Comment by eriwen** [comment]
I like what you're doing to get node support here. 

Could you please provide a bit of the motivation for why it's important to be able to substitute `atob` when we can just use `Buffer`?

**Comment by goldibex** [comment]
In all honesty the double approach was an act of indecision. Overriding `atob` would probably be the more portable option (for Rhino and such). Let me know which way you want it to go and I'll remove the other.

I'll also get the other changes to you ASAP. Thanks again for such a great set of tools; turns out, like Java, Javascript is a "write once, debug everywhere" sort of language.

**Comment by eriwen** [comment]
@goldibex I agree with you that overriding `atob` is better. If you keep with that change we can merge this. I wonder if it'd be worthwhile to provide a `stacktrace-gps` module but for node (this one is really designed for browser use). 

Sorry about the build. We've figured out a way to fix it for PRs but probably won't get it into this library for a few days.

File: spec/stacktrace-gps-spec.js (hunk 2 above)
Comment (goldibex) [inline spec/stacktrace-gps-spec.js]: The test is already checking that the `catch` callback on the promise has been fired. I'll add another one to make sure that the `then` callback was not fired; could you possibly detail what additional assertions you'd like to see here other than that?

File: dist/stacktrace-gps.js (hunk 3 above)
Comment (goldibex) [inline dist/stacktrace-gps.js]: will do!

**Comment by goldibex** [comment]
@eriwen I'm pretty sure that covers all your comments. Let me know if there's anything else.

**Comment by eriwen** [comment]
Looks good, thanks!

**Pull request merged**